### PR TITLE
Fix subbatch loss scaling

### PR DESCRIPTION
**Pull request by kmaziarz**
Currently, the training loss is computed as a _sum_ over sub-batch losses, which itself are _averages_ over contributions from individual samples. This makes the loss scale not invariant to the number of sub-batches, which is not the expected behaviour; moreover, the final sub-batch may end up being smaller, while contributing the same weight into the final loss.

In this PR, I fix the loss scaling so that the loss is _approximately_ invariant to sub-batch size. I also add a test which fails before the fix, and passes after (I've checked the former under 5 random seeds, and the latter under 20 seeds, so the test seems stable despite using `torch.isclose`).

### CODE REVIEW

**Comment by jhrmnn** [comment]
Thanks!
